tls/s2n_handshake_io.c

```diff
@@ -557,6 +605,10 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
     }
 
     if (IS_TLS13_HANDSHAKE(conn)) {
+        if (!conn->quic_enabled) {
+            /* Use middlebox compatibility mode for TLS1.3 by default */
+            conn->handshake.handshake_type |= MIDDLEBOX_COMPAT;
+        }
```

Comment:
Since the default for TLS1.3 is to provide middle box compatible mode, Is there a way to reverse this check from a negative one to a positive test to follow the pattern in pattern in s2n_handshake_io? 

if (IS_TLS13_HANDSHAKE(conn)) {
conn->handshake.handshake_type |= MIDDLEBOX_COMPAT;
}

if (conn->quic_enabled) { 
// Disable Middle Box compatibility mode 
}

This makes it easier to read and follow.

Reply:
I considered that, but I don't think it's actually easier to read. It seems unintuitive to set a flag on one line and then unset it on the very next line. We don't do that for any other handshake flags.

I think there is probably a better way to write that comment tho.